openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/OpenAPIClientGenerator.java

```diff
@@ -151,17 +151,19 @@ public void perform(IDLSourceGeneratorContext idlSourceContext) {
      * This method uses to check whether given specification can be handled via the openapi client generation tool.
      * This includes basic requirements like file extension check and file header check.
      */
-    private static boolean isOpenAPI(Path oasPath) {
+    private static boolean isOpenAPI(Path oasPath, IDLSourceGeneratorContext context) {
 
         try {
             if (!(oasPath.toString().endsWith(YAML_EXTENSION) || oasPath.toString().endsWith(JSON_EXTENSION) ||
                     oasPath.toString().endsWith(YML_EXTENSION))) {
                 return false;
             }
             String content = Files.readString(oasPath);
-            Pattern pattern = Pattern.compile("openapi:");
-            Matcher matcher = pattern.matcher(content);
-            return matcher.find();
+            Pattern swaggerPattern = Pattern.compile("swagger:");
+            Pattern openapiPattern = Pattern.compile("openapi:");
+            Matcher openapiMatcher = openapiPattern.matcher(content);
+            Matcher swaggerMatcher = swaggerPattern.matcher(content);
```

Comment:
Shall we improve the openAPI contract detection by including the version regex as well? That will further reduce the chance of false positives.

Reply:
Yeah here we checked the version of YAML. when it is with `swagger` , it should be version 2. When it starts with `openapi`, it version should be 3 or higher.  :)

Reply:
Yes but still we can use a regex expression to capture the entire version definition line, as the version is semver compatible

Reply:
We can even further improve this logic to use a single regex which can capture both the swagger and openAPI cases.

Reply:
+1 for one regex for addressing both versions. fixed with https://github.com/ballerina-platform/openapi-tools/pull/1086/commits/620687cf94857743cee7d275bfcb4cf0d66a834a